lib/math/math.go

```diff
@@ -0,0 +1,181 @@
+		"sqrt": newUnaryBuiltin("sqrt", math.Sqrt),
+
+		"acos":  newUnaryBuiltin("acos", math.Acos),
+		"asin":  newUnaryBuiltin("asin", math.Asin),
+		"atan":  newUnaryBuiltin("atan", math.Atan),
+		"atan2": newBinaryBuiltin("atan2", math.Atan2),
+		"cos":   newUnaryBuiltin("cos", math.Cos),
+		"hypot": newBinaryBuiltin("hypot", math.Hypot),
+		"sin":   newUnaryBuiltin("sin", math.Sin),
+		"tan":   newUnaryBuiltin("tan", math.Tan),
+
+		"degrees": newUnaryBuiltin("degrees", degrees),
+		"radians": newUnaryBuiltin("radians", radians),
+
+		"acosh": newUnaryBuiltin("acosh", math.Acosh),
+		"asinh": newUnaryBuiltin("asinh", math.Asinh),
+		"atanh": newUnaryBuiltin("atanh", math.Atanh),
+		"cosh":  newUnaryBuiltin("cosh", math.Cosh),
+		"sinh":  newUnaryBuiltin("sinh", math.Sinh),
+		"tanh":  newUnaryBuiltin("tanh", math.Tanh),
+
+		"log": newLogFunction(),
+
+		"gamma": newUnaryBuiltin("gamma", math.Gamma),
+
+		"e":  starlark.Float(math.E),
+		"pi": starlark.Float(math.Pi),
+	},
+}
+
+// floatOrInt is an Unpacker that converts a Starlark int or float to Go's float64.
+type floatOrInt float64
+
+func (p *floatOrInt) Unpack(v starlark.Value) error {
+	switch v := v.(type) {
+	case starlark.Int:
+		*p = floatOrInt(v.Float())
+		return nil
+	case starlark.Float:
+		*p = floatOrInt(v)
+		return nil
+	}
+	return fmt.Errorf("got %s, want float or int", v.Type())
+}
+
+// newUnaryBuiltin wraps a unary floating-point Go function
+// as a Starlark built-in that accepts int or float arguments.
+func newUnaryBuiltin(name string, fn func(float64) float64) *starlark.Builtin {
+	return starlark.NewBuiltin(name, func(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+		var x floatOrInt
+		if err := starlark.UnpackPositionalArgs(name, args, kwargs, 1, &x); err != nil {
+			return nil, err
+		}
+		return starlark.Float(fn(float64(x))), nil
+	})
+}
+
+// newBinaryBuiltin wraps a binary floating-point Go function
+// as a Starlark built-in that accepts int or float arguments.
+func newBinaryBuiltin(name string, fn func(float64, float64) float64) *starlark.Builtin {
+	return starlark.NewBuiltin(name, func(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+		var x, y floatOrInt
+		if err := starlark.UnpackPositionalArgs(name, args, kwargs, 2, &x, &y); err != nil {
+			return nil, err
+		}
+		return starlark.Float(fn(float64(x), float64(y))), nil
+	})
+}
+
+//  newLogFunction wraps the Log function
+// as a Starlark built-in that accepts int or float arguments.
+func newLogFunction() *starlark.Builtin {
+	return starlark.NewBuiltin("log", func(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+		var (
+			x    floatOrInt
+			base = floatOrInt(math.E)
+		)
+		if err := starlark.UnpackArgs("log", args, kwargs, "x", &x, "base?", &base); err != nil {
```

Comment:
UnpackPositionalArgs

(so that the names x and base don't leak into the contract)